library/kotlin/src/io/envoyproxy/envoymobile/filters/Filter.kt

```diff
@@ -39,26 +39,60 @@ internal class EnvoyHTTPFilterAdapter(
     (filter as? ResponseFilter)?.let { responseFilter ->
       val result = responseFilter.onResponseHeaders(ResponseHeaders(headers), endStream)
       return when (result) {
-        is FilterHeadersStatus.Continue -> arrayOf(0 /*EnvoyHTTPHeadersStatusContinue*/, result.headers.headers)
-        is FilterHeadersStatus.StopIteration -> arrayOf(1 /*EnvoyHTTPHeadersStatusStopIteration*/, headers)
+        is FilterHeadersStatus.Continue -> arrayOf(result.status, result.headers.headers)
+        is FilterHeadersStatus.StopIteration -> arrayOf(result.status, headers)
       }
     }
     return arrayOf(0, headers)
   }
 
   override fun onRequestData(data: ByteBuffer, endStream: Boolean): Array<Any> {
+    (filter as? RequestFilter)?.let { requestFilter ->
```

Comment:
I feel like this would be clearer if we just have an if statement
```suggestion
    if (filter is RequestFilter) {
         ...
    }
    return arrayOf(0, data)
```

Reply:
`let` parallels the Swift implementation (and convention) more closely, and I was somewhat under the impression that this was idiomatic(?). But I don't have strong feelings. Just wanted to ask if that swayed our preference at all.